lib/newgistics.rb

```diff
@@ -54,31 +54,19 @@
 require "newgistics/xml_marshaller"
 
 module Newgistics
-  def self.configuration
-    @configuration ||= Configuration.new
-  end
-
-  def self.api
-    @api ||= Api.new
-  end
-
-  def self.logger
-    @logger ||= DefaultLogger.build
-  end
-
-  def self.logger=(logger)
-    @logger = logger
-  end
-
-  def self.time_zone
-    configuration.time_zone
-  end
-
-  def self.local_time_zone
-    configuration.local_time_zone
-  end
-
-  def self.configure
-    yield(configuration)
+  class << self
```

Comment:
I needed `class << self` to get `delegate to work properly. Since I needed it anyway, I put the rest of the methods inside too.